pkg/controller/account/ec2_test.go

```diff
@@ -0,0 +1,301 @@
+package account
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-logr/logr"
+	"github.com/golang/mock/gomock"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/openshift/aws-account-operator/pkg/awsclient"
+	"github.com/openshift/aws-account-operator/pkg/awsclient/mock"
+	"github.com/openshift/aws-account-operator/pkg/controller/testutils"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+type testRunInstanceInputBuilder struct {
+	instanceInput ec2.RunInstancesInput
+}
+
+func newTestRunInstanceInputBuilder() *testRunInstanceInputBuilder {
+	input := ec2.RunInstancesInput{
+		BlockDeviceMappings: []*ec2.BlockDeviceMapping{
+			{
+				DeviceName: aws.String("/dev/sda1"),
+				Ebs: &ec2.EbsBlockDevice{
+					DeleteOnTermination: aws.Bool(true),
+					Encrypted:           aws.Bool(true),
+					VolumeSize:          aws.Int64(10),
+				},
+			},
+		},
+		ImageId:      aws.String("fakeami"),
+		InstanceType: aws.String("t2.micro"),
+		MaxCount:     aws.Int64(1),
+		MinCount:     aws.Int64(1),
+		TagSpecifications: []*ec2.TagSpecification{
+			{
+				ResourceType: &awsv1alpha1.InstanceResourceType,
+				Tags: []*ec2.Tag{
```

Comment:
minor: Could declare the tags array outside this assignment to cut down on length of the builder